WeakAuras/Prototypes.lua

```diff
@@ -4488,59 +4488,88 @@ WeakAuras.event_prototypes = {
     internal_events = {
-      "MAINHAND_TENCH_UPDATE",
-      "OFFHAND_TENCH_UPDATE"
+      "TENCH_UPDATE",
     },
-    force_events = "MAINHAND_TENCH_UPDATE",
-    name = L["Fishing Lure / Weapon Enchant (Old)"],
+    force_events = "TENCH_UPDATE",
+    name = WeakAuras.IsClassic() and L["Weapon Enchant"] or L["Fishing Lure / Weapon Enchant (Old)"],
     init = function(trigger)
       WeakAuras.TenchInit();
+
       local ret = [[
-        local exists, _, name
-        local inverse
-      ]];
-      if(trigger.weapon == "main") then
-        ret = ret .. [[
-          exists, _, name = WeakAuras.GetMHTenchInfo()
-        ]];
-      elseif(trigger.weapon == "off") then
-        ret = ret .. [[
-          exists, _, name = WeakAuras.GetOHTenchInfo()
-        ]];
-      end
+        local triggerWeaponType = %q
+        local triggerName = %q
+        local triggerStack = %s
+        local triggerRemaining = %s
+        local triggerShowOn = %q
+        local _, expirationTime, duration, name, stack, enchantID
 
-      if(trigger.use_inverse) then
-        ret = ret..[[
-          inverse = true;
-        ]];
-      end
+        if triggerWeaponType == "main" then
+          expirationTime, duration, name, shortenedName, _, stack, enchantID = WeakAuras.GetMHTenchInfo()
+        else
+          expirationTime, duration, name, shortenedName, _, stack, enchantID = WeakAuras.GetOHTenchInfo()
+        end
 
-      if(trigger.use_enchant and trigger.enchant and trigger.enchant ~= "") then
-        ret = ret .. [[
-          exists = name == ']] .. trigger.enchant .. [[';
-        ]]
-      end
-      return ret;
+        local remaining = expirationTime and expirationTime - GetTime()
+
+        local nameCheck = triggerName == "" or name and triggerName == name or shortenedName and triggerName == shortenedName or tonumber(triggerName) and enchantID and tonumber(triggerName) == enchantID
+        local stackCheck = not triggerStack or stack and stack %s triggerStack
+        local remainingCheck = not triggerRemaining or remaining and remaining %s triggerRemaining
+        local found = expirationTime and nameCheck and stackCheck and remainingCheck
+      ]];
+
+      return ret:format(trigger.weapon or "main",
+      trigger.use_enchant and trigger.enchant or "",
+      trigger.use_stack and tonumber(trigger.stack or 0) or "nil",
+      trigger.use_remaining and tonumber(trigger.remaining or 0) or "nil",
+      trigger.showOn or "showOnActive",
+      trigger.stack_operator or "<",
+      trigger.remaining_operator or "<")
     end,
     args = {
       {
         name = "weapon",
         display = L["Weapon"],
         type = "select",
         values = "weapon_types",
-        test = "(inverse and not exists) or (not inverse and exists)"
+        test = "true",
+        default = "main",
+        required = true
       },
       {
         name = "enchant",
         display = L["Weapon Enchant"],
+        desc = L["Enchant name or Enchant ID"],
```

Comment:
```suggestion
        desc = L["Enchant Name or ID"],
```

ID(s) if it supports more than one.

Reply:
Weapon Enchantment Name or ID